meson.build

```diff
@@ -0,0 +1,117 @@
+project('profanity',
+        'c',
+        license: 'GPL3+',
+        default_options:  ['c_std=gnu99'],
+        version: '0.12.0',
+        meson_version: '>= 0.54.0')
+
+
+cc = meson.get_compiler('c')
+conf = configuration_data()
+
+
+# general dependencies
+glib_dep = dependency('glib-2.0', version: '>= 2.62.0')
+gio_dep = dependency('gio-2.0')
+curl_dep = dependency('libcurl', version: '>= 7.62.0')
+sqlite_dep = dependency('sqlite3', version: '>= 3.22.0')
+thread_dep = dependency('threads') # pthread
+thread_dep = dependency('readline')
+m_dep = cc.find_library('m', required : true)
+
+
+# this will detect ncursesw etc (-I / -l flags)
+curses_dep = dependency('curses', required: false)
+# get ncurses definitions
+check_headers = [
+  ['ncursesw/menu.h', 'HAVE_NCURSESW_MENU_H'],
+  ['ncurses/menu.h', 'HAVE_NCURSES_MENU_H'],
```

Comment:
You can use `'ncursesw/menu.h'.underscorify().to_upper()` later on in conf.set(), so you only list each header once rather than twice.

Reply:
Thanks for pointing to https://git.sr.ht/~eschwartz/pacman/tree/master/item/meson.build#L123 I think for now I like it like it is :)